tests/rptest/services/keycloak.py

```diff
@@ -0,0 +1,266 @@
+
+    def _create_realm(self, node):
+        rep = {
+            'realm': self.realm,
+            'enabled': True,
+        }
+        cmd = self._construct_admin_cmd([
+            'create realms',
+            '-f -',
+            f"<<< '{json.dumps(rep)}'"
+        ])
+        node.account.ssh(cmd, allow_fail=False)
+
+    def _ssh_decode_json(self, node, cmd, verbose=False, strict=True):
+        bin_out = node.account.ssh_output(cmd, allow_fail=not strict)
+        if verbose:
+            self.logger.info(f"{bin_out}")
+        return json.loads(bin_out.decode('utf-8'))
+
+    def create_client(self, node, client_id, **kwargs):
+        rep = {
+            'clientId': client_id,
+            'enabled': True,
+            'serviceAccountsEnabled': True, # for client credentials grant
+            # 'standardFlowEnabled': True,  # authorization flow grant
+            # 'directAccessGrantsEnabled': True, # resource owner password grant
+            # 'implicitFlowEnabled': True,  # implicit grant
+        }
+        rep.update(kwargs)
+        cmd = self._construct_admin_cmd([
+            'create clients',
+            f'-r {self.realm}',
+            '-f -',
+            f"<<< '{json.dumps(rep)}'"
+        ])
+        out = node.account.ssh_output(cmd, allow_fail=False)
+        self.logger.debug(out)
+
+    def ip(self, node):
+        return node.account.externally_routable_ip
+
+    def get_clients(self, node, realm, client_id=None):
+        args = ['get', 'clients', f'-r {realm}']
+        if client_id is not None:
+            args.append(f'--query clientId={client_id}')
+
+        cmd = self._construct_admin_cmd(args)
+        return self._ssh_decode_json(node, cmd)
+
+    def get_client_uid(self, node, realm, client_id):
+        clients = self.get_clients(node, realm, client_id)
+        if len(clients) == 0:
+            return None
+
+        return clients[0]['id']
+
+    def get_client_secret(self, node, client_id):
+        id = self.get_client_uid(node, self.realm, client_id)
+
+        args = ['create', f'clients/{id}/client-secret', f'-r {self.realm}']
+        node.account.ssh(self._construct_admin_cmd(args), allow_fail=False)
+
+        args[0] = 'get'
+        cmd = self._construct_admin_cmd(args)
+        result = self._ssh_decode_json(node, cmd)
+        return result['value']
+
+    def create_user(self, node, username, password, email):
+        rep = {
+            'username': username,
+            'email': email,
+            'credentials': [{
+                'type': 'password',
+                'value': password,
+                'temporary': False,
+            }],
+            'enabled': True
+        }
+
+        cmd = self._construct_admin_cmd([
+            'create users',
+            f'-r {self.realm}',
+            '-o --fields id,username,clientRoles',
+            f"-b '{json.dumps(rep)}'",
+        ])
+        result = self._ssh_decode_json(node, cmd)
```

Comment:
surprising there isn't a REST interface we can post to directly instead of manually constructing CLI arguments over ssh. in any case, seem fine to me!

Reply:
>surprising there isn't a REST interface we can post to directly

Oh, there is...this app is a thin wrapper over the REST interface with some docs and various convenience parameters. I don't hate it? 

Anyway, this felt a bit more ergonomic starting out, but now knowing the API a bit better could easily convert to bare URLs.

Reply:
Oh, there's a python package. Of course there is 🤦‍♂️